cf/field.py

```diff
@@ -190,11 +191,29 @@
     "__ge__",
 )
 
-_xxx = namedtuple(
-    "data_dimension", ["size", "axis", "key", "coord", "coord_type", "scalar"]
-)
 
-# _empty_set = set()
+@dataclass()
+class _Axis_characterisation:
+    """Characterise a domain axis.
+
+    Used by `_binary_operation` to help with ascertaining if there is
+    a common axis in two fields.
+
+    .. versionaddedd:: NEXTVERSION
+
+    """
+
+    # The size of the axis, a positive integer.
+    size: int = -1
+    # The domain axis identifier. E.g. 'domainaxis0'
+    axis: str = ""
+    # The coordinate constructs that characterize the axis
+    coords: tuple = ()
+    # The identifiers of the coordinate
+    # constructs. E.g. ('dimensioncoordinate1',)
+    keys: tuple = ()
+    # Whether or not the axis is spanned by the field's data array
+    axis_in_data_axes: bool = True
```

Comment:
I'm all for this - please do create that issue.

Reply:
Thanks. Done in #787.

Reply:
I will add equivalent issues to cfdm and cfunits - if we are going to add type hints in, it will be best to start with those smaller (lines of code wise) libraries first, probably.